net/sched: add action inheritdsfield to skbedit

mengxiang0811 opened the pull request:
The new action inheritdsfield copies the field DS of
IPv4 and IPv6 packets into skb->priority. This enables
later classification of packets based on the DS field.

Original idea by Jamal Hadi Salim <jhs@mojatatu.com>

Signed-off-by: Qiaobin Fu <qiaobinf@bu.edu>
Reviewed-by: Michel Machado <michel@digirati.com.br>

AltraMayor commented on tc/m_skbedit.c:
```diff
@@ -111,6 +113,9 @@ parse_skbedit(struct action_util *a, int *argc_p, char ***argv_p, int tca_id,
 			flags |= SKBEDIT_F_PTYPE;
 			ok++;
+		} else if (matches(*argv, "inheritdsfield") == 0) {
+			flags |= SKBEDIT_F_INHERITDSFIELD;
```
Notice that right now, `flags` is not being passed to the kernel.

AltraMayor reviewed (changes requested):
**net/sched: add action inheritdsfield to skbedit**

mengxiang0811 commented:
This patch is ready for another review, and it has been tested.

mengxiang0811 committed:
net/sched: add action inheritdsfield to skbedit

The new action inheritdsfield copies the field DS of
IPv4 and IPv6 packets into skb->priority. This enables
later classification of packets based on the DS field.

v2:
*Use optional flags, so that it won't break old versions of tc.

Original idea by Jamal Hadi Salim <jhs@mojatatu.com>

Signed-off-by: Qiaobin Fu <qiaobinf@bu.edu>
Reviewed-by: Michel Machado <michel@digirati.com.br>
---

Note that the motivation for this patch is found in the following discussion:
https://www.spinics.net/lists/netdev/msg501061.html
